ctapipe/io/simteleventsource.py

```diff
@@ -73,6 +73,27 @@
     -1: EventType.OTHER_CALIBRATION,
 }
 
+_half_pi = 0.5 * np.pi
+_half_pi_maxval = (1 + 1e-6) * _half_pi
+
+
+def _clip_altitude_if_close(altitude):
+    """
+    Round values slightly larger than pi/2 in float64 to pi/2
+
+    These can come from simtel_array because float32(pi/2) > float64(pi/2)
+    and simtel using float32.
+
+    Astropy complains about these values, so we fix them here.
+    """
+    if altitude > _half_pi and altitude < _half_pi_maxval:
```

Comment:
```python 
if _hal_pi < altitude < _half_pi_maxval:
    return _half_pi
```

Reply:
this is *very* slightly slower than the current code.